src/librustdoc/html/render/print_item.rs

```diff
@@ -222,47 +221,83 @@ trait ItemTemplate<'a, 'cx: 'a>: askama::Template + fmt::Display {
     fn item_and_mut_cx(&self) -> (&'a clean::Item, RefMut<'_, &'a mut Context<'cx>>);
 }
 
-fn item_template_document<'a: 'b, 'b, 'cx: 'a>(
-    templ: &'b impl ItemTemplate<'a, 'cx>,
-) -> impl fmt::Display + Captures<'a> + 'b + Captures<'cx> {
-    display_fn(move |f| {
-        let (item, mut cx) = templ.item_and_mut_cx();
-        let v = document(*cx, item, None, HeadingOffset::H2);
-        write!(f, "{v}")
-    })
-}
+macro_rules! item_template {
+    (
+        $(#[$meta:meta])*
+        struct $name:ident<'a, 'cx> {
+            cx: RefCell<&'a mut Context<'cx>>,
+            it: &'a clean::Item,
+            $($field_name:ident: $field_ty:ty),*,
+        },
+        methods = [$($methods:tt),* $(,)?]
```

Comment:
I really like the `methods = ` part. It makes the macro easier to understand when using. Well done! :+1: